modules/selenium/src/test/java/org/testcontainers/junit/ChromeRecordingWebDriverContainerTest.java

```diff
@@ -58,6 +66,52 @@ public String getFilesystemFriendlyName() {
                 assertEquals("Recorded file not found", 1, files.length);
             }
         }
+
+        @Test
+        public void recordingTestThatShouldHaveCorrectDuration() throws IOException, InterruptedException {
+            final String flvFileTitle = "ChromeThatRecordsAllTests-recordingTestThatShouldBeRecordedAndRetained";
+            final String flvFileNameRegEx = "PASSED-" + flvFileTitle + ".*\\.flv";
+            DockerImageName chromeDockerImageName = DockerImageName.parse("selenium/standalone-chrome-debug")
+                                                             .withTag(SeleniumUtils.determineClasspathSeleniumVersion());
+            try (
+                BrowserWebDriverContainer chrome = new BrowserWebDriverContainer(chromeDockerImageName)
+                    .withRecordingMode(RECORD_ALL, vncRecordingDirectory.getRoot())
+                    .withRecordingFileFactory(new DefaultRecordingFileFactory())
+            ) {
+                chrome.start();
+
+                doSimpleExplore(chrome);
+                chrome.afterTest(new TestDescription() {
+                    @Override
+                    public String getTestId() {
+                        return getFilesystemFriendlyName();
+                    }
+
+                    @Override
+                    public String getFilesystemFriendlyName() {
+                        return flvFileTitle;
+                    }
+                }, Optional.empty());
+
+                String recordedFile = vncRecordingDirectory.getRoot().listFiles(new PatternFilenameFilter(flvFileNameRegEx))[0].getCanonicalPath();
+
+                String dockerOutput;
+                try( GenericContainer<?> container = new GenericContainer<>(TestcontainersConfiguration.getInstance().getVncDockerImageName()) ) {
```

Comment:
I wonder, can't we use `VncRecordingContainer` here directly? You can still override `cmdModifier` etc.

Reply:
`VncRecordingContainer` container has constructors with `@NonNull` arguments, and I don't know how can I instantiate the container without creating a method to return a `VncRecordingContainer` with `ffmpeg` as an entrypoint, which I think not good for the class as It will deviate from its main role which is  a vnc recoding, and `ffmpeg` is just an implementation detail (To fix the recording), and in my test I just want to get meta data for the recorded video using whatever `ffmpeg`.

Nonetheless, I can apply your suggestion, I just need some hints on how to do it properly, I don't want to mess the code up, as I don't have a deep knowledge of the base code  :sweat_smile:

Reply:
ok, makes sense.